### ISSUE TITLE
override show_opt_in to false at object level

### ISSUE THREAD

**Issue by meekmeek35** [json-editor/json-editor#992]
### General information

For most of our parameters set in the JSON shema, we want that non-required parameters are displayed with the show_opt_in checkbox. So for this purpose, the show_opt_in option has been set at the jsoneditor level.

But for some few non-required parameters, we would like to remove the show_opt_in checkbox (these are parameters that depends on another value). But defining the show_opt_in to false the object options has no effect.

{
  "type": "object",
  "required": [
    "Mode",
    "RepeatThresholdSecs",
    "ReportPacketCount"
  ],
  "properties": {
    "Mode": {
      "type": "string",
      "enum": [
        "NoAggregation",
        "FullUrl",
        "Hostname",
        "Tag",
        "TagNoUrl"
      ],
      "default": "Tag"
    },
    "RepeatThresholdSecs": {
      "type": "integer",
      "title": "Aggregation time (seconds)",
      "default": 60,
      "minimum": 1
    },
    "ReportPacketCount": {
      "type": "boolean",
      "title": "Report packet count",
      "default": true
    },
    "MinimumReportingCount": {
      "type": "integer",
      "title": "Minimum reporting count",
      "minimum": 0,
      "default": 0,
      "required": [],
      "options": {
        "dependencies": {
          "ReportPacketCount": true
        },
        "show_opt_in": false
      }
    }
  }
}

This could be fixed by changing al the if...else statement on show_opt_in from
if (this.jsoneditor.options.show_opt_in || this.options.show_opt_in)
To
if ((this.options.show_opt_in === undefined && this.jsoneditor.options.show_opt_in) || this.options.show_opt_in) 
so that the show_opt_in option is always used if defined and default to the jsoneditor value if undefined.

- json-editor version: 2.5.4

### Expected behavior

When selecting "ReportPacketCount" to true, JSON editor shall display the MinimumReportingCount parameter without any show_opt_in checkbox.

### Actual behavior

When selecting "ReportPacketCount" to true, JSON editor display the MinimumReportingCount parameter with the show_opt_in checkbox unchecked.

### Steps to reproduce the behavior

On the "https://json-editor.github.io/json-editor/" page, Select the options "Show optional properties (with checkbox)" and use the Json schema above.

Direct link to example: https://json-editor.github.io/json-editor/

```javascript
{
  "type": "object",
  "required": [
    "Mode",
    "RepeatThresholdSecs",
    "ReportPacketCount"
  ],
  "properties": {
    "Mode": {
      "type": "string",
      "enum": [
        "NoAggregation",
        "FullUrl",
        "Hostname",
        "Tag",
        "TagNoUrl"
      ],
      "default": "Tag"
    },
    "RepeatThresholdSecs": {
      "type": "integer",
      "title": "Aggregation time (seconds)",
      "default": 60,
      "minimum": 1
    },
    "ReportPacketCount": {
      "type": "boolean",
      "title": "Report packet count",
      "default": true
    },
    "MinimumReportingCount": {
      "type": "integer",
      "title": "Minimum reporting count",
      "minimum": 0,
      "default": 0,
      "required": [],
      "options": {
        "dependencies": {
          "ReportPacketCount": true
        },
        "show_opt_in": false
      }
    }
  }
}
```

**Comment by schmunk42** [comment]
Yeah, there are editor options which can not be set at the individual editor level.
We had similar issues before be were unable so far to review the code why this happens.

Feedback welcome.

CC: @robocoder

**Comment by meekmeek35** [comment]
In fact in the editor.js and object.js, the show_opt_in is checked like this:
if (this.jsoneditor.options.show_opt_in || this.options.show_opt_in)

So if the jsoneditor is set to true at the high level, there is no way to redefine the show_op_in option at object level to "false" to override the "true" value.

I tested on my side to fix by changing the check to:
if ((this.options.show_opt_in === undefined && this.jsoneditor.options.show_opt_in) || this.options.show_opt_in) 

This is OK for our needs, but i do not know if this is a behavior you want to implement on your side.

Regards
